lib/Request.js

```diff
@@ -124,101 +133,124 @@ Request.prototype.setQuery = function(query, value) {
         }
 
-        var value = this.getQuery(key);
-        if (typeof value === 'undefined' || value === '') {
+        //If null are allowed and the value is null than don't transform the value
+        if (value == null && schemaProperty.type.indexOf('null') !== -1) {
             continue;
         }
 
-        // We have to do isNaN checks here because NaN can cause
-        // false validation success, because:
-        //
-        // 1) (typeof NaN === 'number') evaluates to `true` and
-        // 2) (NaN < 1) evaluates to `false`
-        switch (schema.properties[key].type) {
-            case 'integer':
-                value = parseInt(value);
-                if (isNaN(value)) return typeError(key);
-                break;
-
-            case 'number':
-                value = parseFloat(value);
-                if (isNaN(value)) return typeError(key);
-                break;
+        if (schemaProperty.type.indexOf('integer') !== -1) {
+            value = parseInt(value);
+
+            if (isNaN(value)) {
+                this._validationErrors.push(typeError(key));
+                return;
+            }
         }
 
-        query[key] = value;
-    }
+        if (schemaProperty.type.indexOf('float') !== -1) {
+            value = parseFloat(value);
+            if (isNaN(value)) {
+                this._validationErrors.push(typeError(key));
+                continue;
+            }
+        }
 
-    var valid = validate(query);
-    this.setQueries(query);
+        if (schemaProperty.type.indexOf('boolean') !== -1) {
+            switch(value) {
+                case 'true' :
+                case true :
+                    value = true;
+                    break
+                case 'false':
+                case false:
+                    value = false
+                    break
+                default:
+                    this._validationErrors.push(typeError(key));
+                    continue;
+            }
+        }
 
-    if (true === valid) {
-        return {
-            valid: true,
-            errors: []
-        };
-    } else {
-        return {
-            valid: false,
-            errors: validate.errors
+        if (schemaProperty.type.indexOf('string') !== -1) {
+            value = value.toString();
+        }
+
+        if (schemaProperty.enum && schemaProperty.enum.indexOf(value) === -1) {
+            this._validationErrors.push(typeError(key, 'enum'));
+            continue;
+        }
+
+        object[key] = value;
+        this.setQuery(key, value);
+
+        //If the key is a parameter of this url set the value
+        if (this.getParam(key)) {
+            this.setParam(key, value);
         };
     }
-};
 
-function typeError(key) {
+     // Validate object with schema
+    var validate = jsen(this._schema);
+    var valid = validate(object);
+
+    if (!valid) {
+        this._validationErrors = validate.errors;
```

Comment:
The `for` loop above puts a lot of validation errors into `this._validationErrors`, and then it's entirely replaced here.